src/snowflake/snowpark/dataframe.py

```diff
@@ -1904,6 +1933,12 @@ def join(
                 You can also use ``join_type`` keyword to specify this condition.
                 Note that to avoid breaking changes, currently when ``join_type`` is specified,
                 it overrides ``how``.
+            lsuffix: suffix the overlapping columns of the left DataFrame in the result DataFrame when the left and right DataFrames have overlapping column names.
+            rsuffix: suffix the overlapping columns of the right DataFrame in the result DataFrame when the left and right DataFrames have overlapping column names.
+
+            If both ``lsuffix`` and ``rsuffix`` are empty, the overlapping columns will have random column names in the result DataFrame.
```

Comment:
Let's put this to Note section in the docstring?

Reply:
Added Note